### PR TITLE
[#154] check if uploaded file contains unreadable characters

### LINKED ISSUE #154: [Bug] Uploaded Files aren't checked for encoding or type
Current behavior and impact:
When users upload files, they aren't checked for type or content.
This might also be an "issue" with the text input not being validated.

Steps to reproduce:
1. Generate a file with random content (e.g. with `dd if=/dev/urandom of=largefile.txt bs=100K count=1`)
2. Upload said file in the user code upload menu
Alternatively:
1. Upload a binary from `/usr/bin` in the user code upload menu

Expected behavior:
Text box stays empty, when a file, that isn't a text-file is uploaded.

Acceptance Criteria:
- [ ] Code Text box stays empty, when a file, that isn't a text-file is uploaded.

### PR DESCRIPTION
**Pull request by Straczi**
the regex checks if there are any unreadable characters in the file, tested it out with multiple files and works pretty well (without needing to install additional packages)
closes #154

### CODE REVIEW

**Comment by nikkischnelle** [comment]
lgtm once updated :)

**Comment by Straczi** [comment]
done @nikkischnelle  :+1:

File: frontend/app/src/components/CodeEditor/CodeEditor.tsx [hunk 1]
```diff
@@ -88,11 +89,21 @@ export default function CodeEditor({
...
 			const reader = new FileReader();
+			const isIllegalRegex = /\ufffd/; //check if ureadable character is in file
 			reader.onload = (event) => {
 				const text = event.target?.result as string;
-				setProgram(text);
+				if (!isIllegalRegex.test(text)) {
+					setProgram(text);
+				} else {
+					toast({
+						title: "Whoops, that is not corewar program!",
```
Comment (nikkischnelle) [inline frontend/app/src/components/CodeEditor/CodeEditor.tsx:102]: just noticed, shouldn't this say "that is not _a_ corewar program"?

**Review by 0815Sailsman — APPROVED** [review]
cool 👍🏻

**Pull request merged**